### PR TITLE
More InfoBarPanel API udpates

### PR DESCRIPTION
**Pull request by ranjeshj**
Updating the API to match[ API spec](https://github.com/microsoft/microsoft-ui-xaml-specs/pull/103) one more time. Update naming, and don't ignore some of the values in padding/margins.

### CODE REVIEW

File: dev/InfoBar/InfoBarPanel.cpp [hunk 1]
```diff
@@ -19,20 +19,28 @@ winrt::Size InfoBarPanel::MeasureOverride(winrt::Size const& availableSize)
...
         const auto childDesiredSize = child.DesiredSize();
 
         if (childDesiredSize.Width != 0 && childDesiredSize.Height != 0)
         {
             // Add up the width of all items if they were laid out horizontally
-            const auto horizontalMargin = winrt::InfoBarPanel::GetSpacingInHorizontalOrientation(child);
-            totalWidth += childDesiredSize.Width + (nItems > 0 ? (float)horizontalMargin.Left : 0) + (float)horizontalMargin.Right;
+            const auto horizontalMargin = winrt::InfoBarPanel::GetHorizontalOrientationMargin(child);
+            // Ignore left margin of first and right margin of last child
+            totalWidth += childDesiredSize.Width +
```
Comment (MikeHillberg) [inline dev/InfoBar/InfoBarPanel.cpp:34]: Isn't this supposed to ignore the first/last margins of the items that  aren't collapsed? So we should do a check for non-zero childDesiredSize.Width?

File: dev/InfoBar/InfoBarPanel.cpp [hunk 2]
```diff
@@ -61,18 +69,18 @@ winrt::Size InfoBarPanel::MeasureOverride(winrt::Size const& availableSize)
     if (nItems == 1 || totalWidth > availableSize.Width || (minHeight > 0 && heightOfTallestInHorizontal > minHeight))
     {
         m_isVertical = true;
-        const auto verticalMargin = PaddingInVerticalOrientation();
+        const auto verticalMargin = VerticalOrientationPadding();
```
Comment (MikeHillberg) [inline dev/InfoBar/InfoBarPanel.cpp:72]: Minor: confusing for this variable to be named "margin"

**Review by kmahone — APPROVED** [review]
:shipit:

File: dev/InfoBar/InfoBarPanel.cpp [hunk 3]
```diff
@@ -117,10 +128,10 @@ winrt::Size InfoBarPanel::ArrangeOverride(winrt::Size const& finalSize)
                 auto const desiredSize = child.DesiredSize();
                 if (desiredSize.Width != 0 && desiredSize.Height != 0)
                 {
-                    auto horizontalMargin = winrt::InfoBarPanel::GetSpacingInHorizontalOrientation(child);
+                    auto horizontalMargin = winrt::InfoBarPanel::GetHorizontalOrientationMargin(child);
 
                     horizontalOffset += hasPreviousElement ? (float)horizontalMargin.Left : 0;
-                    child.Arrange(winrt::Rect{ horizontalOffset, (float)horizontalMargin.Top, desiredSize.Width, finalSize.Height });
+                    child.Arrange(winrt::Rect{ horizontalOffset, (float)horizontalOrientationPadding.Top + (float)horizontalMargin.Top, desiredSize.Width, finalSize.Height });
```
Comment (MikeHillberg) [inline dev/InfoBar/InfoBarPanel.cpp:134]: Why do we use desiredSize/finalSize here, but desiredSize/desiredSize for vertical?

File: dev/InfoBar/InfoBarPanel.cpp (hunk 2 above)
Comment (teaP) [reply dev/InfoBar/InfoBarPanel.cpp:72]: Changed to verticalPadding/horizontalPadding.

File: dev/InfoBar/InfoBarPanel.cpp (hunk 3 above)
Comment (teaP) [reply dev/InfoBar/InfoBarPanel.cpp:134]: Not sure.... I switched it to desiredSize for all and can't tell the difference, so hopefully it's OK.

File: dev/InfoBar/InfoBarPanel.cpp (hunk 1 above)
Comment (teaP) [reply dev/InfoBar/InfoBarPanel.cpp:34]: The if statement above this already removes them:
if (childDesiredSize.Width != 0 && childDesiredSize.Height != 0)

File: dev/InfoBar/InfoBarPanel.cpp (hunk 3 above)
Comment (ranjeshj) [reply dev/InfoBar/InfoBarPanel.cpp]: @teaP  can you test to ensure this does not regress behavior ?

Comment (teaP) [reply dev/InfoBar/InfoBarPanel.cpp]: I did, that's what I mean by "can't tell the difference". :D

Comment (MikeHillberg) [reply dev/InfoBar/InfoBarPanel.cpp]: I think finalSize.Height would be the height of the tallest child, so desiredSize.Height sounds right. That also matches StackPanel's code.

Comment (teaP) [reply dev/InfoBar/InfoBarPanel.cpp]: Yep, agreed!

Comment (MikeHillberg) [reply dev/InfoBar/InfoBarPanel.cpp]: (Actually, finalSize.Height should be the height of the tallest child plus the padding?)

**Review by MikeHillberg — APPROVED** [review]

**Pull request merged**